server/server.go

```diff
@@ -398,6 +401,18 @@ func NewServer(configFile string) (*Server, error) {
 			SkipLogging: true,
 		}))
 	}
+	etcdKey := config.GetString("healthcheck/etcd_key", "/gohan")
+	var gohanPort int
+	addressAndPort := strings.Split(server.address, ":")
+	if len(addressAndPort) != 2 {
+		panic("Incorrect gohan server address: " + server.address)
+	}
+	gohanPort, err = strconv.Atoi(addressAndPort[1])
+	if err != nil {
+		panic("Incorrect gohan server address: expected port number got " + addressAndPort[1])
+	}
+	healthCheckAddress := config.GetString("healtcheck/address", fmt.Sprintf("%s:%d", addressAndPort[0], gohanPort+1))
```

Comment:
comment 1: typo `healtcheck`

comment 2: How should we handle the case where healthcheck port is configured as the same value as server port?

Reply:
done, added error